dev/SplitButton/SplitButton.cpp

```diff
@@ -134,7 +134,14 @@ void SplitButton::UpdateVisualStates(bool useTransitions)
     {
         if (m_isFlyoutOpen)
         {
-            winrt::VisualStateManager::GoToState(*this, L"FlyoutOpen", useTransitions);
+            if (InternalIsChecked())
+            {
+                winrt::VisualStateManager::GoToState(*this, L"CheckedFlyoutOpen", useTransitions);
+            }
+            else
+            {
+                winrt::VisualStateManager::GoToState(*this, L"FlyoutOpen", useTransitions);
+            }
```

Comment:
@YuliKl @ranjeshj I've added this condition to apply CheckedFlyoutOpen state when flyout is open with a checked togglesplitbutton because it was not there before.
Whenever flyout opened - it was reverting back to "FlyoutOpen" style which is only right for unchecked scenario.

Please, let me know if that's a known bug and there's a reason why it was not fixed